codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/util/Panic.kt

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rust.codegen.util
+
+fun PANIC(reason: String): Nothing = throw NotImplementedError(reason)
```

Comment:
Should this be called `UNIMPLEMENTED`, and then a separate `PANIC` added that throws `IllegalStateException`?